app/test/account/like_test.dart

```diff
@@ -14,184 +15,232 @@ void main() {
-
-      final rs = await client.listPackageLikes();
-      expect(rs.likedPackages!.isEmpty, true);
-
-      final rs1 = await client2.listPackageLikes();
-      expect(rs1.likedPackages!.isEmpty, true);
-
-      final rs2 = await client.likePackage(pkg1);
-      expect(rs2.package, pkg1);
-      expect(rs2.liked, true);
-
-      final rs3 = await client2.listPackageLikes();
-      expect(rs3.likedPackages!.isEmpty, true);
+      await withHttpPubApiClient(
+        bearerToken: userAtPubDevAuthToken,
+        fn: (client) async {
+          await withHttpPubApiClient(
+            bearerToken: adminAtPubDevAuthToken,
+            fn: (client2) async {
+              final rs = await client.listPackageLikes();
+              expect(rs.likedPackages!.isEmpty, true);
+
+              final rs1 = await client2.listPackageLikes();
+              expect(rs1.likedPackages!.isEmpty, true);
+
+              final rs2 = await client.likePackage(pkg1);
+              expect(rs2.package, pkg1);
+              expect(rs2.liked, true);
+
+              final rs3 = await client2.listPackageLikes();
+              expect(rs3.likedPackages!.isEmpty, true);
+            },
+          );
+        },
+      );
     });
 
     testWithProfile('Package likes property is incremented/decremented.',
         fn: () async {
-      final client = createPubApiClient(authToken: userAtPubDevAuthToken);
-      final client2 = createPubApiClient(authToken: adminAtPubDevAuthToken);
-
-      final rs = await client.getPackageLikes(pkg1);
-      expect(rs.likes, 0);
-      final rs2 = await client2.getPackageLikes(pkg1);
-      expect(rs2.likes, 0);
-
-      await client.likePackage(pkg1);
-
-      final rs3 = await client.getPackageLikes(pkg1);
-      expect(rs3.likes, 1);
-      final rs4 = await client2.getPackageLikes(pkg1);
-      expect(rs4.likes, 1);
-
-      await client2.likePackage(pkg1);
-
-      final rs5 = await client.getPackageLikes(pkg1);
-      expect(rs5.likes, 2);
-      final rs6 = await client2.getPackageLikes(pkg1);
-      expect(rs6.likes, 2);
-
-      await client.unlikePackage(pkg1);
-
-      final rs7 = await client.getPackageLikes(pkg1);
-      expect(rs7.likes, 1);
-      final rs8 = await client2.getPackageLikes(pkg1);
-      expect(rs8.likes, 1);
-
-      /// Unliking already unliked package doesn't cause decrement
-      await client.unlikePackage(pkg1);
-
-      final rs9 = await client.getPackageLikes(pkg1);
-      expect(rs9.likes, 1);
-      final rs10 = await client2.getPackageLikes(pkg1);
-      expect(rs10.likes, 1);
-
-      await client2.unlikePackage(pkg1);
-
-      final rs11 = await client.getPackageLikes(pkg1);
-      expect(rs11.likes, 0);
-      final rs12 = await client2.getPackageLikes(pkg1);
-      expect(rs12.likes, 0);
+      await withHttpPubApiClient(
+        bearerToken: userAtPubDevAuthToken,
+        fn: (client) async {
+          await withHttpPubApiClient(
```

Comment:
Maybe we should tweak `testWithProfile` to create a global `http.Client get testHttpClient` that is created/closed before/after tests are running.

That way we don't have to do: `withHttpPubApiClient`
We can just make a `final client = pubApiClient(bearerToken: ...);`

or something like that. I don't think we need to block this PR.

Reply:
Good idea!